### DISCUSSION TITLE
[BE] id를 인자로 받는 생성자 추가(by 레오)

### DISCUSSION THREAD

**Discussion by yenawee** [woowacourse-teams/2023-stamp-crush#351] [General]
현재 Coupon의 isNotAccessible() 메서드를 보면 customer, cafe의 equals를 이용하는데요. 현재 Customer, Cafe에 equals가 재정의되어 있지 않습니다. 그래서 메모리 주소로 비교하는데 사실, id값으로 비교하도록 재정의해야 합니다. 이렇게 재정의 할 경우 해당 엔티티들도 id를 인자로 받는 생성자가 없기 때문에 테스트가 힘들 것 같아요.

그래서 id를 받는 생성자를 만드는 것도 얘기해보면 좋을 것 같아요

**Comment by jundonghyuk** [comment]
agree with you

**Comment by gitchannn** [comment]
# 결론
## 만드세용~

**Comment by gitchannn** [comment]
단위테스트가 힘들다는거죠?
